Add messages.properties

fortuunn5 opened the pull request:
Add messages.properties, add MessageConfig class

fortuunn5 committed:
Add messages.properties

Add messages.properties, add MessageConfig class

RomanProkhorovich commented on src/main/java/com/example/generatorRandomValues/controller/AdviceController.java:
```diff
@@ -1,18 +1,30 @@
+        return new ResponseEntity<>(message, HttpStatus.BAD_REQUEST);
+    }
+
+    private String getMessage(String messageKey) {
+        Locale locale = Locale.ROOT;
+        return messageSource.getMessage(messageKey, null, locale);
     }
```
вынести в статик сервис

fortuunn5 committed:
Fix AdviceController

fix AdviceController, add MessageUtil

RomanProkhorovich commented on src/main/java/com/example/generatorRandomValues/controller/AdviceController.java:
```diff
@@ -1,18 +1,33 @@
     @ExceptionHandler(UnsupportedTemporalTypeException.class)
-    public ResponseEntity<?> UnsupportedTemporalTypeException(UnsupportedTemporalTypeException e) {
-        return new ResponseEntity<>("Неподходящий шаблон форматирования", HttpStatus.BAD_REQUEST);
+    public ResponseEntity<?> unsupportedTemporalTypeException(UnsupportedTemporalTypeException e) {
+        String message = MessageUtil.getMessage("unsupportedTemporalTypeException.message", messageSource);
+        return new ResponseEntity<>(message, HttpStatus.BAD_REQUEST);
+    }
+
+    private static class MessageUtil {
+        public static String getMessage(String messageKey, MessageSource ms) {
+            Locale locale = Locale.ROOT;
+
+            return ms.getMessage(messageKey, null, locale);
+        }
```
тебе по итогу все равно придется внедрять месседжсорс в класс где ты будешь это использовать. для таких целей ты написала MessageUtil

fortuunn5 committed:
Add static method for message

Add getMessage static method for MessageUtil, add getMessageSourceBean static method for ContextHelper

RomanProkhorovich commented on src/main/java/com/example/generatorRandomValues/utils/ContextHelper.java:
```diff
@@ -48,6 +49,10 @@ static <T extends TypeFilterable> T getBeanByClassForType(Class<T> clazz, @Nonnu
+    public static MessageSource getMessageSourceBean() {
+        return (MessageSource) context.getBean("messageSource");
+    }
+
```
может нпе выпасть

RomanProkhorovich commented on src/main/java/com/example/generatorRandomValues/utils/MessageUtil.java:
```diff
@@ -0,0 +1,16 @@
+    public static String getMessage(String messageKey) {
+        MessageSource messageSource = ContextHelper.getMessageSourceBean();
+        Locale locale = Locale.ROOT;
+        return messageSource.getMessage(messageKey, null, locale);
```
а всегла ли не будет параметров ?

fortuunn5 committed:
Refactor

Add BeanNotFoundException, check NPE
